### PR TITLE
Add Fields to Create Connection Modal and Integrate Backend Workflow

### PR DESCRIPTION
**Pull request by aliraza556**
## Closed: #2159

image

image

image

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device
- [x] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

### CODE REVIEW

**Comment by humansinstitute** [comment]
@tobi-bams can you check this when you come online.

File: db/structs.go [hunk 1]
```diff
@@ -226,6 +226,9 @@ type ConnectionCodes struct {
 	ConnectionString string     `json:"connection_string"`
 	IsUsed           bool       `json:"is_used"`
 	DateCreated      *time.Time `json:"date_created"`
+	Pubkey           string     `json:"pubkey"`
```
Comment (tobi-bams) [inline db/structs.go:229]: Hi @aliraza556 we don't need `Pubkey`, `RouteHint` and `SatsAmount` in the ConnectionCodes struct

File: handlers/auth.go [hunk 2]
```diff
@@ -64,26 +64,34 @@ func (ah *authHandler) CreateConnectionCode(w http.ResponseWriter, r *http.Reque
...
 	}
 
 	if codeBody.RouteHint != "" && codeBody.Pubkey == "" {
-		fmt.Println("pubkey missing missing")
 		w.WriteHeader(http.StatusNotAcceptable)
+		json.NewEncoder(w).Encode("pubkey is required when Route hint is provided")
+		return
+	}
+
+	if codeBody.SatsAmount == 0 {
```
Comment (tobi-bams) [inline handlers/auth.go:92]: It's not an issue if sats amount is zero, we have a default amount to send if `satsAmount` is zero

File: tribes.sql [hunk 3]
```diff
@@ -141,5 +141,8 @@ CREATE TABLE connectioncodes {
   id SERIAL PRIMARY KEY,
   connection_string TEXT,
   is_used boolean,
-  date_created timestamptz
+  date_created timestamptz,
```
Comment (tobi-bams) [inline tribes.sql:144]: Same here we don't need to update the `connectioncodes` table

**Review by tobi-bams — CHANGES_REQUESTED** [review]

File: handlers/auth.go (hunk 2 above)
Comment (aliraza556) [reply handlers/auth.go:92]: @tobi-bams i fix it and then ping you.

**Comment by aliraza556** [comment]
Hi @tobi-bams, Fixed Please review this PR.

**Review by tobi-bams — APPROVED** [review]

**Pull request merged**